client/core/core.go

```diff
@@ -5126,70 +5128,84 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) {
 	var wg sync.WaitGroup
 	conns := c.dexConnections()
 	for _, dc := range conns {
-		if dc.acct.isViewOnly() {
-			continue // don't attempt authDEX for view-only conn
-		}
+		wg.Add(1)
+		go func(dc *dexConnection) {
+			defer wg.Done()
+			c.initializeDEXConnection(dc, crypter)
+		}(dc)
+	}
 
-		// Unlock before checking auth and continuing, because if the user
-		// logged out and didn't shut down, the account is still authed, but
-		// locked, and needs unlocked.
-		err := dc.acct.unlock(crypter)
-		if err != nil {
-			subject, details := c.formatDetails(TopicAccountUnlockError, dc.acct.host, err)
-			c.notify(newFeePaymentNote(TopicAccountUnlockError, subject, details, db.ErrorLevel, dc.acct.host)) // newDEXAuthNote?
-			continue
-		}
+	wg.Wait()
+}
 
-		// Unlock the bond wallet if a target tier is set.
-		if bondAssetID, targetTier, maxBondedAmt := dc.bondOpts(); targetTier > 0 {
-			c.log.Debugf("Preparing %s wallet to maintain target tier of %d for %v, bonding limit %v",
-				unbip(bondAssetID), targetTier, dc.acct.host, maxBondedAmt)
-			wallet, exists := c.wallet(bondAssetID)
-			if !exists || !wallet.connected() { // connectWallets already run, just fail
-				subject, details := c.formatDetails(TopicBondWalletNotConnected, unbip(bondAssetID))
-				var w *WalletState
-				if exists {
-					w = wallet.state()
-				}
-				c.notify(newWalletConfigNote(TopicBondWalletNotConnected, subject, details, db.ErrorLevel, w))
-			} else if !wallet.unlocked() {
-				err = wallet.Unlock(crypter)
-				if err != nil {
-					subject, details := c.formatDetails(TopicWalletUnlockError, dc.acct.host, err)
-					c.notify(newFeePaymentNote(TopicWalletUnlockError, subject, details, db.ErrorLevel, dc.acct.host))
-				}
-			}
-		}
+// initializeDEXConnection connects to the DEX server in the conns map and
+// authenticates the connection.
+func (c *Core) initializeDEXConnection(dc *dexConnection, crypter encrypt.Crypter) {
+	if dc.acct.isViewOnly() {
+		return // don't attempt authDEX for view-only conn
+	}
 
-		if dc.acct.authed() { // should not be possible with newly idempotent login, but there's AccountImport...
-			continue // authDEX already done
-		}
+	if initialized, _ := dc.acct.status(); !initialized {
+		return // dex account is not yet initialized, so we can't unlock it.
+	}
```

Comment:
This isn't necessary. The problem you had before was that you were using `connectDEX` in `ToggleAccountStatus`, and `connectDEX` doesn't set the `connectDEXFlag` so `dc.acct.viewOnly` was not being set correctly. Now that you're  using `connectAccount`, this isn't a problem.

Reply:
I traced your last error report to this: dc.acct.encKey was not set hence the error, which means it hasn't been initialized and this was an attempt to fix that.

Reply:
Please read again and look at the code. Nothing I said was incorrect. Your use of `connectDEX`, which doesn't set the `connectDEXFlag` correctly, was the problem.